pkgs/tools/compression/bzip2/default.nix

```diff
@@ -10,37 +9,41 @@
 
 stdenv.mkDerivation rec {
   pname = "bzip2";
-  version = "1.0.6.0.1";
+  version = "1.0.8";
 
-  /* We use versions patched to use autotools style properly,
-      saving lots of trouble. */
   src = fetchurl {
-    urls = map
-      (prefix: prefix + "/people/sbrabec/bzip2/tarballs/${pname}-${version}.tar.gz")
-      [
-        "http://ftp.uni-kl.de/pub/linux/suse"
-        "ftp://ftp.hs.uni-hamburg.de/pub/mirrors/suse"
-        "ftp://ftp.mplayerhq.hu/pub/linux/suse"
-        "http://ftp.suse.com/pub" # the original patched version but slow
-      ];
-    sha256 = "0b5b5p8c7bslc6fslcr1nj9136412v3qcvbg6yxi9argq9g72v8c";
+    url = "https://sourceware.org/pub/bzip2/${pname}-${version}.tar.gz";
+    sha256 = "0s92986cv0p692icqlw1j42y9nld8zd83qwhzbqd61p1dqbh6nmb";
   };
 
-  nativeBuildInputs = [ autoreconfHook ];
+  outputs = [ "bin" "dev" "out" "man" ] ++ lib.optionals linkStatic [ "static" ];
 
-  patches = [
-    ./CVE-2016-3189.patch
-    ./cve-2019-12900.patch
-  ];
-
-  postPatch = ''
-    sed -i -e '/<sys\\stat\.h>/s|\\|/|' bzip2.c
+  # fix shared library options
+  postPatch = stdenv.lib.optionalString stdenv.hostPlatform.isDarwin ''
+    substituteInPlace Makefile-libbz2_so \
+        --replace "soname" "install_name"
```

Comment:
This isn't enough, the install_name should point to the absolute path of the library not just it's basename since darwin doesn't use rpath.